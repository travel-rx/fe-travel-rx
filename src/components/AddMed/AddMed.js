import React, { Component } from 'react';
import { StyleSheet, Text, View, ScrollView, TextInput, TouchableOpacity, Picker, Dimensions } from 'react-native';
import Footer from '../Footer/Footer';
import { RFPercentage } from "react-native-responsive-fontsize";

const { height, width } = Dimensions.get('screen');

export default class AddMed extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      dosage: '',
      frequency: 1,
      withFood: false,
      inventory: 0
    }
  }

  static navigationOptions = {
    title: 'Add Medication',
    headerLeft: null,
    headerStyle: {
      backgroundColor: '#3499AA'
    },
    headerTintColor: '#EBEBEB',
    headerTitleStyle: {
      fontSize: height * 0.05,
    },
  };

  render() {
    const { navigation } = this.props;
    return(
      <View style={styles.container}>
        <ScrollView>
          <View style={styles.addMed}>
            <TextInput
              style={styles.input}
              placeholder='Medication name'
              textAlign='center'
              onChangeText={(name) => this.setState({ name })}
              value={this.state.name}
              keyboardShouldPersistTaps='always'
            />
            <TextInput
              style={styles.input}
              placeholder='Dosage'
              textAlign='center'
              onChangeText={(dosage) => this.setState({ dosage })}
              value={this.state.dosage}
              keyboardShouldPersistTaps='always'
            />
            <TextInput 
              style={styles.input}
              placeholder='Times taken per day'
              textAlign='center'
              keyboardType={'numeric'}
              onChange={(frequency) => this.setState({ frequency })}
              value={this.state.frequency}
              keyboardShouldPersistTaps='always'
            />
            <Picker
              selectedValue={this.state.withFood}
              style={styles.picker}
              onValueChange={(itemValue) => this.setState({ withFood: itemValue})}
            >
              <Picker.Item label='Take WITH food' value={false}/>
              <Picker.Item label='Take WITHOUT food' value={true}/>
            </Picker>
            <TouchableOpacity style={styles.button}>
              <Text style={styles.text}>Save</Text>
            </TouchableOpacity>
          </View>
        </ScrollView>
        <Footer navigation={navigation}/>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  text: {
    color: '#EBEBEB',
    textAlign: 'center',
    fontSize: height * 0.04
  },
  input: {
    borderColor: '#D7D7D7',
    borderRadius: 12,
    borderWidth: 2,
    height: height * 0.08,
    fontSize: 15,
    marginTop: 20,
    fontSize: RFPercentage(4),
    width: width * 0.85,
  },
  button: {
    backgroundColor: '#3499AA',
    borderColor: 'white',
    borderWidth: 1,
    borderRadius: 12,
    display: 'flex',
    fontWeight: 'bold',
    height: height * 0.06,
    justifyContent: 'center',
    marginTop: 75,
    overflow: 'hidden',
    width: width * 0.85,
  },
  addMed: {
    marginTop: 30
  },
  picker: {
    flex: 0,
    justifyContent: 'space-around',
    height: 75,
    marginTop: 50
  }
});