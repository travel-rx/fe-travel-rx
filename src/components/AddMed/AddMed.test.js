import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import AddMed from './AddMed';
import { shallow } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('AddMed', () => {
  it('should return true', () => {
    expect(true).toEqual(true);
  });

  it('should match the snapshot', () => {
    expect(shallow(<AddMed />)).toMatchSnapshot();
  })
})