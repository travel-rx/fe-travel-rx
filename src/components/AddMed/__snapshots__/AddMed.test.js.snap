// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMed should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <ScrollViewMock>
    <View
      style={
        Object {
          "marginTop": 30,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        keyboardShouldPersistTaps="always"
        onChangeText={[Function]}
        placeholder="Medication name"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#D7D7D7",
            "borderRadius": 12,
            "borderWidth": 2,
            "fontSize": 53,
            "height": 106.72,
            "marginTop": 20,
            "width": 637.5,
          }
        }
        textAlign="center"
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        keyboardShouldPersistTaps="always"
        onChangeText={[Function]}
        placeholder="Dosage"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#D7D7D7",
            "borderRadius": 12,
            "borderWidth": 2,
            "fontSize": 53,
            "height": 106.72,
            "marginTop": 20,
            "width": 637.5,
          }
        }
        textAlign="center"
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        keyboardShouldPersistTaps="always"
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="Times taken per day"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#D7D7D7",
            "borderRadius": 12,
            "borderWidth": 2,
            "fontSize": 53,
            "height": 106.72,
            "marginTop": 20,
            "width": 637.5,
          }
        }
        textAlign="center"
        underlineColorAndroid="transparent"
        value=""
      />
      <Picker
        mode="dialog"
        onValueChange={[Function]}
        selectedValue={false}
        style={
          Object {
            "flex": 0,
            "height": 75,
            "justifyContent": "space-around",
            "marginTop": 50,
          }
        }
      >
        <PickerItem
          label="Take WITH food"
          value={false}
        />
        <PickerItem
          label="Take WITHOUT food"
          value={true}
        />
      </Picker>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#3499AA",
            "borderColor": "white",
            "borderRadius": 12,
            "borderWidth": 1,
            "display": "flex",
            "fontWeight": "bold",
            "height": 80.03999999999999,
            "justifyContent": "center",
            "marginTop": 75,
            "overflow": "hidden",
            "width": 637.5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#EBEBEB",
              "fontSize": 53.36,
              "textAlign": "center",
            }
          }
        >
          Save
        </Text>
      </TouchableOpacity>
    </View>
  </ScrollViewMock>
  <Footer />
</View>
`;
